src/cascadia/TerminalConnection/ConptyConnection.cpp

```diff
@@ -149,9 +144,13 @@ namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
             // WSLENV is a colon-delimited list of environment variables (+flags) that should appear inside WSL
             // https://devblogs.microsoft.com/commandline/share-environment-vars-between-wsl-and-windows/
 
-            auto wslEnv = environment[L"WSLENV"];
-            wslEnv = L"WT_SESSION:" + wslEnv; // prepend WT_SESSION to make sure it's visible inside WSL.
-            environment.insert_or_assign(L"WSLENV", wslEnv);
+            const auto existingWslEnv = environment.as_map()[L"WSLENV"];
+            std::wstring wslEnv{ L"WT_SESSION:WT_PROFILE_ID" };
```

Comment:
now here's a curious question.

Say somebody has a WSL profile with a special `environment`. _It's not going to work" unless they also add a `WSLENV` mentioning every other var they added.

Should we do that for them? What's the least surprising thing here? /cc @zadjii-msft 

On one side: `wsl profile -> environment` doesn't work properly
On the other: terminal might defy simple expectations

Reply:
So, if we do this... we should probably _append_ our WSLENV instead of prepend it. Because that will allow the user to specify the different WSLENV flags and not have us overwrite them.

@benhill, if Terminal emits a duplicate entry in `WSLENV`, will WSL be OK? Is the order of `WSLENV` parsing guaranteed? That is, if we happen to emit `WSLENV=X/p:X` will `X` be guaranteed to be parsed ala `/p`?

Reply:
I'm sorry, other Ben Hill. I think I do that to you all the time. I switched over to the @benhillis I was after. ☹️

Reply:
Okay, empirically observed. We should prepend to `WSLENV`.

```
# Pretend that we inserted our entries here        |vvv|
(dhowett-sl) ~ % $Env:A; $Env:B; $Env:WSLENV="A/p:B:A:B"; wsl --% echo $A $B
c:\foo
foobar
c:\foo foobar
```

`A` did not get `/p` path translation.

```
# Instead, we insert our entries here        |vvv|
(dhowett-sl) ~ % $Env:A; $Env:B; $Env:WSLENV="A:B:A/p:B"; wsl --% echo $A $B
c:\foo
foobar
/mnt/c/foo foobar
```

It worked!